Fill project settings tab

jcheringer opened the pull request:
## Summary 
This PR will start filling the project's sidebar settings with information we already have on the API.

Ref: c/Qga7WIeF-tr

![image](https://user-images.githubusercontent.com/7811225/144642404-f66ce6ed-9c36-4f10-9ba7-4543b9c03e86.png)

## Test Instructions

* Checkout this branch
* Run `yarn install` and then start the dashboard running `yarn workspace @crowdsignal/dashboard start`
* Open or create a new project
* Click the settings icon on the upper right corner and select `Document` tab
* You should something similar with the image above

jcheringer commented:
@ice9js and @CGastrell though we might need some more work on this, I'm adding you as a reviewer so that we can start talking a look at the things in the current stage.

ice9js commented on apps/dashboard/src/components/editor/document-settings.js:
```diff
@@ -3,13 +3,62 @@
+			</PanelBody>
+			<PanelBody title={ __( 'Permalink', 'dashboard' ) }>
+				<PanelRow>
+					<span>View Project</span>
```
Missing translation.

ice9js commented on apps/dashboard/src/components/editor/document-settings.js:
```diff
@@ -3,13 +3,62 @@
+			</PanelBody>
+			<PanelBody title={ __( 'Permalink', 'dashboard' ) }>
+				<PanelRow>
+					<span>View Project</span>
```
Shouldn't this be a link/button?

jcheringer replied:
Considering the image attached to the card and the equivalent Permalink implementation on WP.com, this seems to be plain text.

ice9js reviewed (commented):
Looks good - I only found a missing translation and I also think the `GlobalStylesUI` comments can be removed for now. I'm not sure that's where these belong when we implement them anyway.

Other than that, I'd love to see some helper functions for reading some of the project's values, but that's somewhat addressed by #75 which is currently on hold and we'll be dealing with it while we're in beta stage. So I think it's best to just leave that as is and refactor everything together later.

jcheringer commented:
@ice9js thanks for reviewing it!
I've addressed the comments and, when testing the latest changes, I noticed some required styles adjustments.
Please take a look at the styles changes and let me know if there are any necessary changes to make it more consistent with the project design.

CGastrell commented:
Weird, followed the instructions and I'm seeing this:
![image](https://user-images.githubusercontent.com/157240/145121666-f322783e-6e51-4cd5-9e8c-8f4545bd1724.png)

Did I miss something?

jcheringer commented:
> Weird, followed the instructions and I'm seeing this: ![image](https://user-images.githubusercontent.com/157240/145121666-f322783e-6e51-4cd5-9e8c-8f4545bd1724.png)
> 
> Did I miss something?

Well, this is what I see too.
Considering what we have for now, this is the result I got.

What is bugging you? Maybe I am missing something 😅

CGastrell commented:
> What is bugging you? Maybe I am missing something

LOL! I saw the design and I thought it was your screenshot, sorry

digitalwaveride commented:
Nice João!!! 👍

Three little things:
- The tab itself should be called "Project" not "Document".
- Let's skip the "Last" and just call it "Updated", if there is no other way to avoid the linebreak.
- It seems like there is a little space in front of the "minutes":
<img width="231" alt="image" src="https://user-images.githubusercontent.com/5120357/145272052-3a7bc443-17a8-4fdc-9b31-9a6166a4f69c.png">

Thanks so much! 🙏

jcheringer commented:
> Nice João!!! 👍
> 
> Three little things:
> 
> * The tab itself should be called "Project" not "Document".
> * Let's skip the "Last" and just call it "Updated", if there is no other way to avoid the linebreak.
> * It seems like there is a little space in front of the "minutes":
> 
> <img alt="image" width="231" src="https://user-images.githubusercontent.com/5120357/145272052-3a7bc443-17a8-4fdc-9b31-9a6166a4f69c.png">
> 
> Thanks so much! 🙏

Thanks for taking a look Thomas!

Regarding the tab name, I changed the [settings.js](https://github.com/Automattic/crowdsignal-ui/pull/101/files#diff-b7e6c9db5640ef744abed57d40c1f531122b24a98e5b27ffbe7166a89b70ddea) and I'm pretty sure I have seen this text changed during the development. Not sure why it's not working. Any insights @ice9js ?

About the `Last Updated`, to avoid the line break, other than skipping the "Last", the only alternatives I can see are: use an abbreviated DateTime or increase the sidebar width (I guess it's possible).

> * It seems like there is a little space in front of the "minutes":

For this one, indeed, there is a space between the minutes and seconds. I got inspiration from [PostScheduleLabel](https://github.com/WordPress/gutenberg/blob/9df736c678b2e8ef00c355e6cc9fb8092fe25484/packages/editor/src/components/post-schedule/label.js#L8) and it uses  the `@wordpress/date` package, which has this (don't ask me why) [space between the minutes and seconds](https://github.com/WordPress/gutenberg/blob/trunk/packages/date/src/index.js#L125)

As an alternative, we could create our own `DATETIME_FORMAT` constant and make it the format we want to show.

Please, let me know your thoughts. Thanks again! 😃

digitalwaveride commented:
Let's skip the "Last" for now, simplest solution.

We should keep the date-format that WP uses and not change it for now.

Regarding the space, I am not seeing an issue on WP.com editor... not sure what they are using... 🤷 

<img width="294" alt="image" src="https://user-images.githubusercontent.com/5120357/145277560-f7d050d9-558a-4dd0-a346-bfdbfc9ad80f.png">

jcheringer commented:
> Let's skip the "Last" for now, simplest solution.
> 
> We should keep the date-format that WP uses and not change it for now.
> 
> Regarding the space, I am not seeing an issue on WP.com editor... not sure what they are using... 🤷
> 
> <img alt="image" width="294" src="https://user-images.githubusercontent.com/5120357/145277560-f7d050d9-558a-4dd0-a346-bfdbfc9ad80f.png">

Pushed some changes to adjust the "Last Update" label and the date-time format.
About the space between the hour and minutes, I tried to get some help from @CGastrell and, from our understanding the previous implementation was using the same rules WP.com uses. We were not able to identify why the results on our side are different.
This way, I ended up creating our own date-time formatting string. I'm keeping the same format WP.com uses but, this way, I was able to remove the unnecessary space.

digitalwaveride commented:
Thanks João for looking into the date format! Really appreciated!

I am not sure, maybe just me, but I am seeing this error here:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/5120357/145306706-578b25d5-2f32-4c24-ba3b-3f3a48d8c098.png">

jcheringer commented:
> Thanks João for looking into the date format! Really appreciated!
> 
> I am not sure, maybe just me, but I am seeing this error here:
> 
> <img alt="image" width="1792" src="https://user-images.githubusercontent.com/5120357/145306706-578b25d5-2f32-4c24-ba3b-3f3a48d8c098.png">

I'm not seeing this error in my environment.
Maybe @ice9js can take a look at his env and let us know? 🙏

ice9js commented on apps/dashboard/src/components/editor/document-settings.js:
```diff
@@ -3,13 +3,60 @@
+import { ExternalLink, PanelBody, PanelRow } from '@wordpress/components';
+
+const DocumentSettings = ( { project } ) => {
+	const DATETIME_FORMAT = 'F j, Y H:i';
```
NIt: this constant and `formatDate` could be defined outside of `DocumentSettings` since they don't actually rely on anything from the component in their implementation. Let's just put them above it.

ice9js commented on apps/dashboard/src/components/editor/settings.js:
```diff
@@ -29,7 +29,7 @@ export const editorSettings = {
 		},
 		toolbar: {
-			documentInspector: __( 'Document', 'dashboard' ),
+			documentInspector: __( 'Project', 'dashboard' ),
```
Technically, this doesn't seem to be valid configuration anymore and the recent version of Automattic/isolated-block-editor expects a bool.  
Let's leave it for now and raise an issue there.

ice9js reviewed (changes requested):
I'm not seeing any errors either and am able to test things.

Regarding the document tab name, it seems that [when sidebars were introduced to the isolated block editor](https://github.com/Automattic/isolated-block-editor/commit/38ef55f33e3a9f9ab89af18654db04b097629760) the `documentInspector` setting was made into a boolean and the title is now [hardcoded to 'Document'](https://github.com/Automattic/isolated-block-editor/blob/trunk/src/components/block-editor/sidebar-heading.js#L17).  
Unfortunately there's nothing we can do about that in the very short term so I'd consider that outside of the scope for this PR from now on.

I think the permalink could be made to look a little bit better if we add these styles to the wrapper `<a>`:

```css
text-overflow: ellipsis;
max-width: 100%;
box-sizing: border-box;
overflow: hidden;
display: inline-block;
```

Before:
![Screen Shot 2021-12-10 at 2 32 05 PM](https://user-images.githubusercontent.com/8056203/145581849-4d02c3b3-9df7-4b1d-90c4-256ab049112a.png)

After:
![Screen Shot 2021-12-10 at 2 29 27 PM](https://user-images.githubusercontent.com/8056203/145581775-e1c348b7-d062-4491-b3be-c2f3f131a67d.png)

---

I left another small comment as well. Let's get these changes in and merge it.

jcheringer commented:
> I'm not seeing any errors either and am able to test things.
> 
> Regarding the document tab name, it seems that [when sidebars were introduced to the isolated block editor](https://github.com/Automattic/isolated-block-editor/commit/38ef55f33e3a9f9ab89af18654db04b097629760) the `documentInspector` setting was made into a boolean and the title is now [hardcoded to 'Document'](https://github.com/Automattic/isolated-block-editor/blob/trunk/src/components/block-editor/sidebar-heading.js#L17). Unfortunately there's nothing we can do about that in the very short term so I'd consider that outside of the scope for this PR from now on.
> 
> I think the permalink could be made to look a little bit better if we add these styles to the wrapper `<a>`:
> 
> ```css
> text-overflow: ellipsis;
> max-width: 100%;
> box-sizing: border-box;
> overflow: hidden;
> display: inline-block;
> ```
> 
> Before: ![Screen Shot 2021-12-10 at 2 32 05 PM](https://user-images.githubusercontent.com/8056203/145581849-4d02c3b3-9df7-4b1d-90c4-256ab049112a.png)
> 
> After: ![Screen Shot 2021-12-10 at 2 29 27 PM](https://user-images.githubusercontent.com/8056203/145581775-e1c348b7-d062-4491-b3be-c2f3f131a67d.png)
> 
> I left another small comment as well. Let's get these changes in and merge it.

Thanks for reviewing it again @ice9js!
I've addressed the comments and also used the opportunity to try cleaning the `document-settings.js` up a bit.

ice9js reviewed (approved):
LGTM! Thanks!
